yarn-project/aztec-node/src/aztec-node/http-node.ts

```diff
@@ -0,0 +1,305 @@
+import { AztecNode } from '@aztec/aztec-node';
+import {
+  AztecAddress,
+  CONTRACT_TREE_HEIGHT,
+  Fr,
+  KernelCircuitPublicInputs,
+  L1_TO_L2_MESSAGES_TREE_HEIGHT,
+  PRIVATE_DATA_TREE_HEIGHT,
+  Proof,
+  PublicCallRequest,
+  SignedTxRequest,
+} from '@aztec/circuits.js';
+import { SiblingPath } from '@aztec/merkle-tree';
+import {
+  ContractData,
+  ContractPublicData,
+  EncodedContractFunction,
+  L2Block,
+  MerkleTreeId,
+  Tx,
+  TxHash,
+  UnverifiedData,
+} from '@aztec/types';
+
+/**
+ * Serialises a transaction to JSON representation.
+ * @param tx - The transaction to serialise.
+ * @returns The serialsied transaction.
+ */
+export function txToJson(tx: Tx) {
+  return {
+    data: tx.data?.toBuffer().toString('hex'),
+    unverified: tx.unverifiedData?.toBuffer().toString('hex'),
+    txRequest: tx.txRequest?.toBuffer().toString('hex'),
+    proof: tx.proof?.toBuffer().toString('hex'),
+    newContractPublicFunctions: tx.newContractPublicFunctions?.map(f => f.toBuffer().toString('hex')) ?? [],
+    enqueuedPublicFunctions: tx.enqueuedPublicFunctionCalls?.map(f => f.toBuffer().toString('hex')) ?? [],
+  };
+}
```

Comment:
You can pass in a custom serializer object rather than manually converting every buffer to string. See the implementation of `toFriendlyJSON` for an example.

And aside from that, why not use the serialization methods we already have, and push everything over the wire as binary data?

Reply:
I just used strings for ease of debugging. Being able to get legible output by cURLing the endpoints is pretty useful. We probably want to change this for at least some endpoints for a production system.